.github/workflows/hackage.yml

```diff
@@ -111,7 +112,7 @@ jobs:
           echo "allow-newer: Chart-diagrams:diagrams-core, SVGFonts:diagrams-core," >> cabal.project
 
       - name: "Add temporary needed allow-newer for ghc-9.0"
```

Comment:
Just while I'm here: for this and the above step, don't we already have modified `cabal.project` files for exactly this kind of thing?

Reply:
well, it is a long story:
* we have to avoid any cabal.project tweaks (`allow-newer`, `source-repository-packages`) as hackage doesnt see them and without them the hackage build fails
* we want to make the packages tests and benchmarks work in hackage
* cabal solver needs all packages (including tests and benchmarks as we want enable them in general) be "solvable" although you dont build them effectively (like brittany for ghc-9.0.1)

* so we have to replicate the minimum possible of cabal.project tweaks in a controlled way to make the build work

But maybe there is another way to do it, o revise the initial premises

Reply:
Right, but this is for GHC 9, right? Where we *do* have `cabal.project` tweaks. Basically, I don't see how adding them here can be any better than adding them in `cabal-ghc9.project` or whatever?

Reply:
Hmm, not sure if i am understanding, those tweaks are already in cabal-ghc901.project but we dont want to use the entire file, only the things needed by the solver which we know will not make the hackage build fail.
For example we are not building brittany for ghc-9, neither in the project nor in hackage so it is safe to add the `allow-newer`'s of brittany to make happy the solver cause the hackage build for ghc-9 of brittany is disabled by flags. 

Another alternative could be comment out the not buildable packages (like we do temporary, see for example the actual `cabal-ghc921.project`) but it would make more complex the actual `echo "packages: . ../../* ../../plugins/*" > cabal.project`

Reply:
Ah I see, I think I just didn't understand the point either. Let me think about it a bit, and anyway it's irrelevant to this PR :)

Reply:
Just a *nota bene*: the https://github.com/haskell/haskell-language-server/pull/2567/files#r779448472 is indeed a complex reason & seems like a case for a code comment/doc "why it was needed here".

Reply:
Reported in https://github.com/haskell/haskell-language-server/issues/2586